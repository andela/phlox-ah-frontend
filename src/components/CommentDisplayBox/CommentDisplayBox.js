import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';
import Avatar from 'react-avatar';
import './CommentDisplayBox.scss';

const CommentDisplayBox = ({ comment }) => (
  <div className="comment-display">
    <div className="author-details">
      <div className="author-image">
        {
          comment.User.Profile.profileImage ? <img className="profileImage" src={comment.User.Profile.profileImage} />
            : <Avatar name={comment.User.username} size="50" round={true} />
        }
      </div>
      <div className="more-details">
        <span className="author-name">{comment.User.Profile.firstName} {comment.User.Profile.lastName}</span>
        <span className="date">{moment(comment.createdAt).format('ll')}</span>
      </div>
      <div className="user-options">
        <i className="fas fa-edit edit"></i>
        <i className="fas fa-trash delete"></i>
      </div>
    </div>

    <p className="comment-content">{comment.comment}</p>

    <div className="comment-like">
      <span className="likes">
        <i className="fas fa-thumbs-up"></i>
        <span className="likes-count">10</span>
      </span>
      <span className="likes">
        <i className="fas fa-thumbs-down"></i>
        <span className="likes-count">23</span>
      </span>
    </div>

    <div className="user-reply">
      <div className="reply-box">
        <div className="user-reply-image"></div>
        <input
          type="text"
          name="reply"
          className="reply-input"
          placeholder="Add a reply here..." />
      </div>
      <div className="reply-button">
        <button>Reply</button>
      </div>
    </div>
  </div>
);

CommentDisplayBox.propTypes = {
  comment: PropTypes.object
};

export default CommentDisplayBox;
