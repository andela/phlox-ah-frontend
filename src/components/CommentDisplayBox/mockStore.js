const comment = {
  Replies: [],
  User: {
    Profile: {
      firstName: 'Jame',
      lastName: 'Jackson',
      profileImage: null
    },
    email: 'admin@authorshaven.com',
    username: 'admin1'
  },
  articleSlug: 'title-of-article2',
  comment: 'good work',
  createdAt: '2018-11-21T08:25:35.435Z',
  id: 3,
  likes: [],
  updatedAt: '2018-11-21T08:25:35.435Z',
  userId: 6
};

export default comment;
