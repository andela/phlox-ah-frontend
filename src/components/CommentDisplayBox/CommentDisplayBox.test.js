import React from 'react';
import { mount } from 'enzyme';
import CommentDisplayBox from './CommentDisplayBox';
import comment from './mockStore';

let component;

beforeEach(() => {
  component = mount(<CommentDisplayBox comment={comment} />);
});

afterEach(() => {
  component.unmount();
});

it('should render without throwing an error', () => {
  expect(component).toMatchSnapshot();
});

it('should have a div', () => {
  expect(component.find('div').length).toEqual(8);
});

it('should have a span', () => {
  expect(component.find('span').length).toEqual(8);
});

it('should have a p', () => {
  expect(component.find('p').length).toEqual(1);
});

it('should have a i', () => {
  expect(component.find('i').length).toEqual(2);
});

it('should have a props of comment that is an object', () => {
  expect(typeof component.props().comment).toBe('object');
});

it('should have a first name and last name', () => {
  expect(component.find('.author-name').text()).toEqual('Jame Jackson');
});

it('should have a comment', () => {
  expect(component.find('.comment-content').text()).toEqual('good work');
});

it('should have a data', () => {
  expect(component.find('.date').text()).toEqual('Nov 21, 2018');
});
