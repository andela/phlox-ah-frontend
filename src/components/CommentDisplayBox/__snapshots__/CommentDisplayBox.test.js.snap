// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render without throwing an error 1`] = `
<CommentDisplayBox
  comment={
    Object {
      "Replies": Array [],
      "User": Object {
        "Profile": Object {
          "firstName": "Jame",
          "lastName": "Jackson",
          "profileImage": null,
        },
        "email": "admin@authorshaven.com",
        "username": "admin1",
      },
      "articleSlug": "title-of-article2",
      "comment": "good work",
      "createdAt": "2018-11-21T08:25:35.435Z",
      "id": 3,
      "likes": Array [],
      "updatedAt": "2018-11-21T08:25:35.435Z",
      "userId": 6,
    }
  }
>
  <div
    className="comment-display"
  >
    <div
      className="author-details"
    >
      <div
        className="author-image"
      >
        <withAvatarConfig
          name="admin1"
          round={true}
          size="50"
        >
          <Avatar
            avatarRedirectUrl={null}
            cache={
              Object {
                "get": [Function],
                "hasSourceFailedBefore": [Function],
                "set": [Function],
                "sourceFailed": [Function],
              }
            }
            className=""
            colors={
              Array [
                "#d73d32",
                "#7e3794",
                "#4285f4",
                "#67ae3f",
                "#d61a7f",
                "#ff4080",
              ]
            }
            fgColor="#FFF"
            initials={[Function]}
            name="admin1"
            round={true}
            size="50"
            textSizeRatio={3}
            unstyled={false}
          >
            <div
              className=" sb-avatar"
              style={
                Object {
                  "borderRadius": "100%",
                  "display": "inline-block",
                  "fontFamily": "Helvetica, Arial, sans-serif",
                  "height": "50px",
                  "verticalAlign": "middle",
                  "width": "50px",
                }
              }
            >
              <div
                className=" sb-avatar__text"
                style={
                  Object {
                    "background": undefined,
                    "borderRadius": "100%",
                    "color": "#FFF",
                    "height": "50px",
                    "lineHeight": "initial",
                    "textAlign": "center",
                    "textTransform": "uppercase",
                    "width": "50px",
                  }
                }
              >
                <div
                  style={
                    Object {
                      "display": "table",
                      "height": "100%",
                      "width": "100%",
                    }
                  }
                >
                  <span
                    style={
                      Object {
                        "display": "table-cell",
                        "verticalAlign": "middle",
                      }
                    }
                  >
                    <span
                      key="null"
                    />
                  </span>
                </div>
              </div>
            </div>
          </Avatar>
        </withAvatarConfig>
      </div>
      <div
        className="more-details"
      >
        <span
          className="author-name"
        >
          Jame Jackson
        </span>
        <span
          className="date"
        >
          Nov 21, 2018
        </span>
      </div>
    </div>
    <p
      className="comment-content"
    >
      good work
    </p>
    <div
      className="comment-like"
    >
      <span
        className="likes"
      >
        <i
          className="fas fa-thumbs-up"
        />
        <span
          className="likes-count"
        >
          10
        </span>
      </span>
      <span
        className="likes"
      >
        <i
          className="fas fa-thumbs-down"
        />
        <span
          className="likes-count"
        >
          23
        </span>
      </span>
    </div>
  </div>
</CommentDisplayBox>
`;
